File: samples/client/petstore/csharp-dotnet2/SwaggerClientTest/Lib/SwaggerClient/src/main/CsharpDotNet2/IO/Swagger/Model/AllOfSubCategoryCategory.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class AllOfSubCategoryCategory : Category {
    /// <summary>
    /// Gets or Sets Foo
    /// </summary>
    [DataMember(Name="foo", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "foo")]
    public bool? Foo { get; set; }

    /// <summary>
    /// Gets or Sets Bar
    /// </summary>
    [DataMember(Name="bar", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "bar")]
    public int? Bar { get; set; }

    /// <summary>
    /// Gets or Sets Beer
    /// </summary>
    [DataMember(Name="beer", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "beer")]
    public string Beer { get; set; }

    /// <summary>
    /// Gets or Sets Drunk
    /// </summary>
    [DataMember(Name="drunk", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "drunk")]
    public User Drunk { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class AllOfSubCategoryCategory {\n");
      sb.Append("  Foo: ").Append(Foo).Append("\n");
      sb.Append("  Bar: ").Append(Bar).Append("\n");
      sb.Append("  Beer: ").Append(Beer).Append("\n");
      sb.Append("  Drunk: ").Append(Drunk).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public  new string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: samples/client/petstore/dart/swagger/lib/model/all_of_sub_category_category.dart
part of swagger.api;

class AllOfSubCategoryCategory {
  
  int id = null;

  String name = null;

  bool foo = null;

  int bar = null;

  String beer = null;

  User drunk = null;

  AllOfSubCategoryCategory();

  @override
  String toString() {
    return 'AllOfSubCategoryCategory[id=$id, name=$name, foo=$foo, bar=$bar, beer=$beer, drunk=$drunk, ]';
  }

  AllOfSubCategoryCategory.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    id = json['id'];
    name = json['name'];
    foo = json['foo'];
    bar = json['bar'];
    beer = json['beer'];
    drunk = new User.fromJson(json['drunk']);
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'foo': foo,
      'bar': bar,
      'beer': beer,
      'drunk': drunk
     };
  }

  static List<AllOfSubCategoryCategory> listFromJson(List<dynamic> json) {
    return json == null ? new List<AllOfSubCategoryCategory>() : json.map((value) => new AllOfSubCategoryCategory.fromJson(value)).toList();
  }

  static Map<String, AllOfSubCategoryCategory> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, AllOfSubCategoryCategory>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new AllOfSubCategoryCategory.fromJson(value));
    }
    return map;
  }
}
